### PR TITLE
Sub tree fixes for issues found on Jarvis

### PR DESCRIPTION
**Pull request by trishaanand**
They should only have property paths and not entity names themselves.

### CODE REVIEW

File: app/client/src/workers/evaluationUtils.ts [hunk 1]
```diff
@@ -297,3 +297,13 @@ export function getValidatedTree(
...
 }
+
+export const isChildPropertyPath = (
+  parentPropertyPath: string,
+  childPropertyPath: string,
+): boolean => {
+  const regexTest = new RegExp(
+    `^${parentPropertyPath.replace(".", "\\.")}(\\.\\S+)?$`,
+  );
+  return regexTest.test(childPropertyPath);
```
Comment (ofpiyush) [inline app/client/src/workers/evaluationUtils.ts:308]: Did you mean `startsWith` here?

Comment (trishaanand) [reply app/client/src/workers/evaluationUtils.ts:308]: No. We are actually doing a proper regex test and testing with the regex is a match or not (true or false). Starts with is just one of the scenarios that the regex covers.

Comment (trishaanand) [reply app/client/src/workers/evaluationUtils.ts:308]: StartsWith also covers scenarios where you are looking for Table1.selectedRow and you mistakenly find Table1.selectedRows. This is incorrect.

Comment (ofpiyush) [reply app/client/src/workers/evaluationUtils.ts:308]: the `.` at the end prevents the case

File: app/client/src/workers/evaluationUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { isChildPropertyPath } from "./evaluationUtils";
+
+describe("isChildPropertyPath function", () => {
+  it("works", () => {
+    const cases: Array<[string, string, boolean]> = [
+      ["Table1.selectedRow", "Table1.selectedRows", false],
+      ["Table1.selectedRow", "Table1.selectedRow.email", true],
+      ["Table1.selectedRow", "1Table1.selectedRow", false],
+      ["Table1.selectedRow", "Table11selectedRow", false],
```
Comment (trishaanand) [inline app/client/src/workers/evaluationUtils.test.ts:9]:
```suggestion
      ["Table1.selectedRow", "Table11selectedRow", false],
      ["Table1.selectedRow", "Table1.selectedRow", true],
```

**Review by ofpiyush — APPROVED** [review]

**Pull request merged**